hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -33,53 +33,82 @@
   public static final String METADATA_PREFIX = "hoodie.metadata";
 
   // Enable the internal Metadata Table which saves file listings
-  public static final String METADATA_ENABLE_PROP = METADATA_PREFIX + ".enable";
-  public static final boolean DEFAULT_METADATA_ENABLE = false;
+  public static final ConfigOption<Boolean> METADATA_ENABLE_PROP = ConfigOption
+      .key(METADATA_PREFIX + ".enable")
+      .defaultValue(false)
+      .withDescription("Enable the internal Metadata Table which stores table level metadata such as file listings");
```

Comment:
Lets just say in the description that it `stores file listings` to avoid confusion. When later its extended for other kinds of metadata we can change.